### PR TITLE
New Integration - AWS EKS

### PR DESCRIPTION
**Pull request by RotemAmit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-7998)

## Description
New Integration - AWS EKS.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+        logging_arg: The cluster control plane logging configuration.
+        authentication_mode: The desired authentication mode for the cluster.
+
+    Returns:
+        A Command Results object
+    """
+    arr = [resources_vpc_config, logging_arg, authentication_mode]
+    arg_num = 0
+    for arg in arr:
+        arg_num = arg_num + 1 if arg else arg_num
```
Comment (eyalpalo) [inline Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py:45]:
```suggestion
    arg_num = sum(bool(arg) for arg in arr) # counts the number of non-empty args.
```

File: Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS_description.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+### Configure AWS Settings
+- Create an IAM role with either of the following IAM policies:
+  - AmazonEKSClusterPolicy
+  - A custom IAM policy. For more information about creating a custom IAM policy see the [Amazon EKS cluster IAM role](https://docs.aws.amazon.com/eks/latest/userguide/service_IAM_role.html).
+- Create an Amazon EKS clusters.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS_description.md:5]:
```suggestion
- Create Amazon EKS clusters.
```

File: Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+''' MAIN FUNCTION '''
+
+
+def main():  # pragma: no cover
+    params = demisto.params()
+    aws_default_region = params.get('defaultRegion')
+    aws_access_key_id = params.get('credentials', {}).get('identifier')
+    aws_secret_access_key = params.get('credentials', {}).get('password')
+    verify_certificate = not demisto.params().get('insecure', False)
+    demisto.params().get('proxy', False)
```
Comment (eyalpalo) [inline Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py:465]: No need, handled by default in the aws client

File: Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.yml [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+  advanced: true
+  required: false
+- display: Trust any certificate (not secure)
+  name: insecure
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  type: 8
+  required: false
```
Comment (eyalpalo) [inline Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.yml:51]: Generally we add also timeout params - please see other aws integration add align the params accordingly.

File: Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.yml [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+      type: String
+    - contextPath: AWS.EKS.UpdateCluster.type
+      description: The type of the update.
+      type: String
+    - contextPath: AWS.EKS.UpdateCluster.params
+      description: The parameters of the update.
+      type: Object
+    - contextPath: AWS.EKS.UpdateCluster.createdAt
+      description: The creation date ob the object.
+      type: Unknown
```
Comment (eyalpalo) [inline Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.yml:154]: String?

**Comment by ShirleyDenkberg** [comment]
@eyalpalo @sapirshuker Doc review completed.

@RotemAmit Make sure to regenerate the integration readme after implementing my comments in the yml file.

File: Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py [hunk 6]
```diff
@@ -0,0 +1,514 @@
...
+    """
+    arr = [resources_vpc_config, logging_arg, authentication_mode]
+    arg_num = 0
+    for arg in arr:
+        arg_num = arg_num + 1 if arg else arg_num
+    if arg_num != 1:
+        raise ValueError("Please provide exactly one of the following arguments: resources_vpc_config, logging or "
+                         "authentication_mode.")
+    else:
+        return 'ok'
```
Comment (eyalpalo) [inline Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py:50]:
```suggestion
    return True
```
Or we can remove it entirely

File: Packs/AWS-EKS/Integrations/AWSEKS/README.md [hunk 7]
```diff
@@ -0,0 +1,466 @@
...
+
+1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
+2. Search for AWS-EKS.
+3. Click **Add instance** to create and configure a new integration instance.
+
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- |--------------|
+    | AWS Default Region |  | True         |
+    | Access Key |  | True         |
+    | Secret Key |  | True        |
```
Comment (eyalpalo) [inline Packs/AWS-EKS/Integrations/AWSEKS/README.md:14]: Please add description

File: Packs/AWS-EKS/Integrations/AWSEKS/README.md [hunk 8]
```diff
@@ -0,0 +1,466 @@
...
+    | Retries | The maximum number of retry attempts when connection or throttling errors are encountered. Set to 0 to disable retries. The default value is 5 and the limit is 10. Note: Increasing the number of retries will increase the execution time. | False        |
+    | Trust any certificate (not secure) |  | False        |
+    | Use system proxy settings |  | False        |
```
Comment (eyalpalo) [inline Packs/AWS-EKS/Integrations/AWSEKS/README.md:17]: Same as above

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Nice! See my comments, some of them appear once or twice but should be implemented on all of the same code in other commands.

File: Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py [hunk 9]
```diff
@@ -0,0 +1,514 @@
...
+    Args:
+        args (dict): A dictionary containing the configuration parameters for the session.
+                     - 'region' (str): The AWS region.
+
+        aws_client (AWSClient): The AWS client used to configure the session.
+
+    Returns:
+        AWS session (boto3 client): The configured AWS session.
+    """
+    return aws_client.aws_session(service='eks', region=args.get('region'))
```
[the remark this answers is not in the document]
Comment (RotemAmit) [reply Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py:67]: It stays to be the same as declared in the params

File: Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py [hunk 10]
```diff
@@ -0,0 +1,514 @@
...
+            ).get('accessEntry')
+        else:
+            response = aws_client.create_access_entry(
+                clusterName=cluster_name,
+                principalArn=principal_arn,
+                kubernetesGroups=kubernetes_groups,
+                tags=tags,
+                clientRequestToken=client_request_token,
+                type=type_arg
+            ).get('accessEntry')
```
[the remark this answers is not in the document]
Comment (RotemAmit) [reply Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py:276]: If I send the username as None, there is an error saying that username should be of type string.
When I send the username as an empty string, I receive an error that the username must contain at least one char.
This is why I did it this way.

File: Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py [hunk 11]
```diff
@@ -0,0 +1,514 @@
...
+                logging=logging_arg,
+            )
+        else:  # access_config
+            response = aws_client.update_cluster_config(
+                name=cluster_name,
+                accessConfig=access_config
+            )
+
+        response_data = response.get('update', {})
+        response_data['clusterName'] = cluster_name
```
[the remark this answers is not in the document]
Comment (eyalpalo) [reply Packs/AWS-EKS/Integrations/AWSEKS/AWSEKS.py:152]: you are right, my bad

**Review by eyalpalo — APPROVED** [review]
Great!

**Pull request merged**